tracker: enable strictDeps

erdnaxe opened the pull request:
###### Description of changes

Motivation: https://github.com/NixOS/nixpkgs/issues/178468

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

jtojnar commented on pkgs/development/libraries/tracker/default.nix:
```diff
@@ -46,7 +46,9 @@ stdenv.mkDerivation rec {
     pkg-config
   ];
 
+  strictDeps = true;
```
Not sure what is the proper place but probably before first/after last dependency list, rather than in the middle. We currently [do not have it mapped](https://github.com/jtojnar/nixpkgs-hammering/blob/main/explanations/attribute-ordering.md).

erdnaxe replied:
Moved, thanks!

erdnaxe commented on pkgs/development/libraries/tracker/default.nix:
```diff
@@ -46,7 +46,9 @@ stdenv.mkDerivation rec {
 
+  strictDeps = true;
   nativeBuildInputs = [
+    dbus
```
Done! It turned out that I missed the fact that the error happens during the check phase, so I moved `dbus` to `nativeCheckInputs` which seems to make more sense.

jtojnar commented:
Oh, and also it should target staging branch due to number of rebuilds.

erdnaxe committed:
tracker: enable strictDeps

Enable strictDeps and fix "dbus-run-session: command not found" error
during test phase.

jtojnar reviewed (approved):
LGTM, thanks.

erdnaxe commented:
> Oh, and also it should target staging branch due to number of rebuilds.

Ooops, I rebased the branch on staging before telling github to target the staging branch, I might have unintentionally pinged some maintainers :/

mweinelt commented:
Just remember to set the PR to draft status before starting the rebase. That will prevent unrelated codeowners from getting requested.
